### ISSUE TITLE
Hangs with React 16.6.0

### ISSUE THREAD

**Issue by sheepsteak** [planttheidea/react-rendered-size#3]
I've found that using the latest React version that was released a few days ago (16.6.0) breaks this module. It seems to get stuck on the `while` loop (https://github.com/planttheidea/react-rendered-size/blob/master/src/element.js#L79) waiting for the render callback to fire.

I think what is happening is that the latest version of `react-dom` is more async and uses something like `requestIdleCallback` to schedule the render. However, this module is hammering the main thread with a `while` loop so the component never gets rendered and the callback never fires.

Only way I can see to fix this is to accept that this can no longer be a synchronous render and use a callback/Promise to wait for the render to finish.

**Comment by planttheidea** [comment]
Agreed ... the unfortunate thing is that this limits the number of potential use-cases (the example App being one of them). I'm going to experiment with some alternative techniques (trampolining, for example) that may allow synchronous renders to continue to be used before I succumb to making this async-only.
